provider/internal/java/dependency.go

```diff
@@ -405,36 +405,60 @@ func (p *javaServiceClient) parseDepString(dep, localRepoPath, pomPath string) (
 	} else {
 		return d, fmt.Errorf("unable to split dependency string %s", dep)
 	}
-	d.Name = fmt.Sprintf("%s.%s", parts[0], parts[1])
 
-	var fp string
-	if d.Classifier == "" {
-		fp = filepath.Join(localRepoPath, strings.Replace(parts[0], ".", "/", -1), parts[1], d.Version, fmt.Sprintf("%v-%v.jar.sha1", parts[1], d.Version))
-	} else {
-		fp = filepath.Join(localRepoPath, strings.Replace(parts[0], ".", "/", -1), parts[1], d.Version, fmt.Sprintf("%v-%v-%v.jar.sha1", parts[1], d.Version, d.Classifier))
+	group := parts[0]
+	artifact := parts[1]
+	d.Name = fmt.Sprintf("%s.%s", group, artifact)
+
+	fp := resolveDepFilepath(&d, p, group, artifact, localRepoPath)
+
+	d.Labels = addDepLabels(p.depToLabels, d.Name)
+	d.FileURIPrefix = fmt.Sprintf("file://%v", filepath.Dir(fp))
+
+	d.Extras = map[string]interface{}{
+		groupIdKey:    group,
+		artifactIdKey: artifact,
+		pomPathKey:    pomPath,
 	}
+
+	return d, nil
+}
+
+// resolveDepFilepath tries to extract a valid filepath for the dependency with either JAR or POM packaging
+func resolveDepFilepath(dep *provider.Dep, p *javaServiceClient, group string, artifact string, localRepoPath string) string {
+	groupPath := strings.Replace(group, ".", "/", -1)
+
+	// Try jar packaging
+	fp := getFilepathForPackaging(dep, localRepoPath, groupPath, artifact, "jar")
 	b, err := os.ReadFile(fp)
+	if err != nil {
+		// Try pom packaging (see https://www.baeldung.com/maven-packaging-types#4-pom)
+		fp := getFilepathForPackaging(dep, localRepoPath, groupPath, artifact, "pom")
+		b, err = os.ReadFile(fp)
+	}
+
 	if err != nil {
 		// Log the error and continue with the next dependency.
-		p.log.V(5).Error(err, "error reading SHA hash file for dependency", "dep", d.Name)
+		p.log.V(5).Error(err, "error reading SHA hash file for dependency", "dep", dep.Name)
 		// Set some default or empty resolved identifier for the dependency.
-		d.ResolvedIdentifier = ""
+		dep.ResolvedIdentifier = ""
 	} else {
 		// sometimes sha file contains name of the jar followed by the actual sha
 		sha, _, _ := strings.Cut(string(b), " ")
-		d.ResolvedIdentifier = sha
+		dep.ResolvedIdentifier = sha
 	}
 
-	d.Labels = addDepLabels(p.depToLabels, d.Name)
-	d.FileURIPrefix = fmt.Sprintf("file://%v", filepath.Dir(fp))
+	return fp
+}
 
-	d.Extras = map[string]interface{}{
-		groupIdKey:    parts[0],
-		artifactIdKey: parts[1],
-		pomPathKey:    pomPath,
+func getFilepathForPackaging(dep *provider.Dep, localRepoPath string, groupPath string, artifact string, packaging string) string {
```

Comment:
Is this method used anywhere else? I would just as soon put this function inline, just reduces the need to jump around to different methods to understand what is happening and it is not that much.